### PR TITLE
Add Java Resources

### PR DESCRIPTION
**Pull request by koustuvsinha**
Add Java Resources as per #534 

@Rafase282 @alayek please review!

### CODE REVIEW

File: Java-Resources.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+- [Project Euler](https://projecteuler.net/)
+- [Practice It! - Java Challenges](http://practiceit.cs.washington.edu/)
+
+
+## Community
+
+- [Awesome Java](https://github.com/akullpp/awesome-java)
+- [/r/Java](https://www.reddit.com/r/Java)
+- [/r/LearnJava](https://www.reddit.com/r/learnjava)
+- [Oracle Java Community](http://www.oracle.com/technetwork/java/community/index.html)
```
Comment (alayek) [inline Java-Resources.md]: Put this at the top. For trademark reasons.

File: Java-Resources.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Java Resources
+
```
Comment (alayek) [inline Java-Resources.md]: Add something like
>  Java is trademarked and licensed by Oracle. Most of the following are un-official resources; unless hosted by Oracle.

File: Java-Resources.md (hunk 1 above)
Comment (alayek) [inline Java-Resources.md]: Add [JCG](https://www.javacodegeeks.com/).

**Comment by Rafase282** [comment]
@koustuvsinha make the changes suggested by @alayek and we are golden.

**Comment by koustuvsinha** [comment]
@Rafase282 updated

File: Java-Resources.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+- [Java 8 Tutorial](https://github.com/winterbe/java8-tutorial)
+- [Better Java](https://github.com/cxxr/better-java)
+- [Java Programming notes by NTU](http://www3.ntu.edu.sg/home/ehchua/programming/index.html#Java)
+- [Princeton's Introduction to Programming in Java](http://introcs.cs.princeton.edu/java/home/)
+- [Introduction to Programming Using Java](http://math.hws.edu/javanotes/)
+- [Java Practices](http://javapractices.com/home/HomeAction.do)
+- [Java Design Patterns](https://github.com/iluwatar/java-design-patterns/)
+- [Algorithms in Java](https://github.com/pedrovgs/Algorithms)
+- [Spring Official Guides](https://spring.io/guides)
+
```
Comment (sandhya-rayaprolu) [inline Java-Resources.md]: I found these sites helpful too

http://www.learnjavaonline.org/ 
programmingbydoing.com
http://www.programcreek.com/java-tutorials/

jpassion.com is a very good resource, but there is a subscription fee of $139/year, so I don't know if that fits in with FCC or not. Their online codecamps are free to attend, paying members get access to materials
http://www.jpassion.com/portal/announcements/online-codecamps

File: Java-Resources.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+## Challenges
+
+- [Java Koans](https://github.com/matyb/java-koans)
+- [Coding Bat Java Challenges](http://codingbat.com/java)
+- [Excercism Java Challenges](http://exercism.io/languages/java)
+- [Project Euler](https://projecteuler.net/)
+- [Practice It! - Java Challenges](http://practiceit.cs.washington.edu/)
+
```
Comment (sandhya-rayaprolu) [inline Java-Resources.md]: https://leetcode.com/ can be added to this list.

**Pull request merged**